pkg/querier/tenantfederation/merge_queryable_test.go

```diff
@@ -33,7 +40,30 @@ func (m *mockTenantQueryableWithFilter) Querier(ctx context.Context, _, _ int64)
 	if err != nil {
 		return nil, err
 	}
-	return mockTenantQuerier{tenant: tenantIDs[0], extraLabels: m.extraLabels}, nil
+
+	q := mockTenantQuerier{tenant: tenantIDs[0], extraLabels: m.extraLabels}
+
+	// set warning if exists
+	if intf := ctx.Value(contextKeyWarningsByTenant); intf != nil {
```

Comment:
Nit: I don't quite see why warnings and errors for tenants must be passed to `mockTenantQueryableWithFilter` via context, instead of having extra fields in that queryable.

Reply:
Valid point, have refactored the tests a bit in 4a10de0